src/main/java/seedu/contax/model/OnboardingStory.java

```diff
@@ -20,4 +24,33 @@ public OnboardingStep getStep(int index) {
     public int getSize() {
         return story.size();
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof OnboardingStory)) {
+            return false;
+        }
+        OnboardingStory otherStory = (OnboardingStory) other;
+
+        if (otherStory.getSize() != getSize()) {
+            return false;
+        }
+
+        for (int i = 0; i < getSize(); i++) {
+            if (!otherStory.getStep(i).equals(getStep(i))) {
+                return false;
+            }
+        }
```

Comment:
You can simplify this to 
```suggestion
        otherStory.story.equals(story);
```

Reply:
Thanks for the suggestion. Fixed.